pkg/resources/cruisecontrol/configmap.go

```diff
@@ -43,11 +44,10 @@ func (r *Reconciler) configMap(clientPass, capacityConfig string, log logr.Logge
 	}
 	ccConfig.Merge(conf)
 
-	// Set bootstrap servers
-	bootstrapServers := fmt.Sprintf("%s:%d",
-		generateBootstrapServer(r.KafkaCluster.Spec.HeadlessServiceEnabled, r.KafkaCluster.Name),
-		generateBootstrapServerPort(log, r.KafkaCluster.Spec.ListenersConfig.InternalListeners),
-	)
+	bootstrapServers := kafkautils.GetBootstrapServers(r.KafkaCluster)
+	if bootstrapServers == "" {
+		log.Error(errors.New("getting bootstrap servers returned an empty string"), "")
+	}
```

Comment:
I would handle this in the GetBootstrapServers() func itself.

Reply:
Alright. Will change the `GetBootstrapServer()` to return an error in case it was not able to determine the Kafka bootstrap server from the provided CR.